File: algorithm-programmers-hash-marathon/src/test/java/com/currenjin/SolutionTest.java
package com.currenjin;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {
    @Test
    void sample1() {
        assertEquals("leo", new Solution().solution(new String[]{"leo", "kiki", "eden"}, new String[]{"eden", "kiki"}));
    }

    @Test
    void duplicateName() {
        assertEquals("mislav", new Solution().solution(
                new String[]{"mislav", "stanko", "mislav", "ana"},
                new String[]{"stanko", "ana", "mislav"}
        ));
    }
}

File: algorithm-programmers-hash-marathon/src/test/kotlin/com/currenjin/SolutionTest.kt
package com.currenjin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class SolutionTest {
    @Test
    fun sample1() {
        val participant = arrayOf("leo", "kiki", "eden")
        val completion = arrayOf("eden", "kiki")
        assertEquals("leo", Solution().solution(participant, completion))
    }

    @Test
    fun duplicateName() {
        val participant = arrayOf("mislav", "stanko", "mislav", "ana")
        val completion = arrayOf("stanko", "ana", "mislav")
        assertEquals("mislav", Solution().solution(participant, completion))
    }
}
